business_requirement_deliverable/models/business.py

```diff
@@ -0,0 +1,398 @@
+# Copyright 2016-2019 Elico Corp (https://www.elico-corp.com).
+# Copyright 2019 Tecnativa - Alexandre Díaz
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.addons import decimal_precision as dp
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class BusinessRequirementDeliverable(models.Model):
+    _name = "business.requirement.deliverable"
+    _inherit = ["mail.thread", "mail.activity.mixin", "portal.mixin"]
+    _description = "Business Requirement Deliverable"
+    _order = "sequence, id"
+
+    sequence = fields.Integer('Sequence')
+    state = fields.Selection(
+        related='business_requirement_id.state',
+        selection=[('draft', 'Draft'),
+                   ('confirmed', 'Confirmed'),
+                   ('approved', 'Approved'),
+                   ('in_progress', 'In progress'),
+                   ('done', 'Done'),
+                   ('cancel', 'Cancel'),
+                   ('drop', 'Drop'),
+                   ],
+        store=True,
+        readonly=True,
+    )
+    name = fields.Text('Name', required=True)
+    user_case = fields.Html()
+    proposed_solution = fields.Html()
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        domain=[('sale_ok', '=', True)],
+        required=False
+    )
+    uom_id = fields.Many2one(
+        comodel_name='product.uom',
+        string='UoM',
+        required=True,
+        default=lambda self: self.env.ref('product.product_uom_unit')
+    )
+    qty = fields.Float(
+        string='Quantity',
+        store=True,
+        default=1,
+    )
+    business_requirement_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Business Requirement',
+        ondelete='cascade',
+        required=True
+    )
+    sale_price_unit = fields.Float(
+        string='Sales Price',
+        oldname='unit_price'
+    )
+    price_total = fields.Float(
+        compute='_compute_get_price_total',
+        string='Total Deliverable',
+        store=True,
+        readonly=True
+    )
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        string='Currency',
+        readonly=True,
+        compute='_compute_get_currency',
+    )
+    business_requirement_partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        related='business_requirement_id.partner_id',
+        string='Stakeholder',
+        store=True
+    )
+    state = fields.Selection(related='business_requirement_id.state',
+                             string='State', store=True, readonly=True)
+    portal_published = fields.Boolean('In Portal', default=True)
```

Comment:
Why not use website_published and inherit from website.published.mixin

Reply:
Because with that, it will depend on `website` module, which is a heavy dependency.